cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -970,6 +988,32 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 		return err
 	}
 
+	// for most cases we allow the use of fallback to unlock/mount things
+	allowFallback := true
+
+	tryingCurrentSystem, err := boot.InitramfsIsTryingRecoverySystem(mst.recoverySystem)
+	if err != nil {
+		if boot.IsInconsystemRecoverySystemState(err) {
+			// there is some try recovery system state in bootenv
+			// but it is inconsistent, make sure we clear it and
+			// return back to run mode
+			finalizeErr := finalizeTryRecoverySystemAndReboot(boot.TryRecoverySystemOutcomeInconsistent)
+			// not reached, unless in tests
+			panic(fmt.Errorf("inconsistent tried recovery system bootenv: %v", finalizeErr))
```

Comment:
s/tried/try?

Reply:
my 2cts is that it was be nicer if finalizeTryRecoverySystemAndReboot would not return an error but either do its job or panic directly

Reply:
fair point, i've refactored the code to end in `finalizeTry..`